spec/migrations/20241017013023_reencrypt_password_scramsha_spec.rb

```diff
@@ -5,18 +5,31 @@
 describe ReencryptPasswordScramsha do
   migration_context :up do
     it "Ensures that the user password is stored as scram-sha-256" do
-      migrate
+      allow(ActiveRecord::Base.connection).to receive(:execute).and_call_original
 
       username = ActiveRecord::Base.connection_db_config.configuration_hash[:username]
 
-      users_and_passwords = ActiveRecord::Base.connection.execute <<-SQL
-        SELECT rolname, rolpassword FROM pg_authid WHERE rolcanlogin;
-      SQL
+      expect(ActiveRecord::Base.connection_db_config).to receive(:configuration_hash).exactly(10).times.and_call_original
```

Comment:
This was an interesting find.

Reply:
Don't love adding "real" code only to support tests, but maybe worth it in this case.  You could add a method that returns the connection_db_config in the migration class and stub that method in this spec

Reply:
Why exactly 10 times as opposed to something like at least once?

Reply:
I'm guessing because on the 11th time he needs to return a different payload